packages/react/src/components/sideNavigation/SideNavigation.tsx

```diff
@@ -0,0 +1,168 @@
+}: SideNavigationProps) => {
+  const container = React.useRef<HTMLDivElement>(null);
+  const customThemeClass = useTheme<SideNavigationCustomTheme>(styles.sideNavigation, theme);
+  const menuId = `${id}-menu`;
+  const [openMainLevels, setOpenMainLevels] = React.useState<number[]>(defaultOpenMainLevels ?? []);
+  const [mobileMenuOpen, setMobileMenuOpen] = React.useState(false);
+  const isMobile = useMobile();
+
+  const mainLevels = React.Children.map(children, (child, index) => {
+    if (isValidElement(child) && (child.type as FCWithName).componentName === 'MainLevel') {
+      return cloneElement(child, { index });
+    }
+    return null;
+  });
+
+  const toggleMobileMenu = () => {
+    setMobileMenuOpen(!mobileMenuOpen);
+  };
+
+  // Handle additional event listeners.
+  // TODO: Add a reusable hook to implement this behaviour
+  useEffect(() => {
+    // Close with ESC key
+    const handleEscKey = (event: KeyboardEvent) => {
+      const key = event.key || event.keyCode;
+      if (mobileMenuOpen && (key === 'Escape' || key === 'Esc' || key === 27)) {
+        setMobileMenuOpen(false);
+      }
+    };
+    // Close with click or focusin outside the component
+    const handleOutsideClickOrFocusin = (event: Event) => {
+      const target = event.target as Node;
+      if (mobileMenuOpen && !container.current.contains(target)) {
+        setMobileMenuOpen(false);
+      }
+    };
+
+    document.addEventListener('click', handleOutsideClickOrFocusin);
+    document.addEventListener('focusin', handleOutsideClickOrFocusin);
+    document.addEventListener('keyup', handleEscKey);
+    return () => {
+      document.removeEventListener('click', handleOutsideClickOrFocusin);
+      document.removeEventListener('focusin', handleOutsideClickOrFocusin);
+      document.removeEventListener('keyup', handleEscKey);
+    };
+  });
+
+  return (
+    <SideNavigationContext.Provider
+      value={{
+        mobileMenuOpen,
+        openMainLevels,
+        setMobileMenuOpen,
+        setOpenMainLevels,
+        autoCollapseOthers,
+      }}
+    >
+      <nav
+        className={classNames(styles.sideNavigation, customThemeClass, className)}
+        id={id}
+        ref={container}
+        style={style}
+      >
+        {/* Toggle button is visible only on small screen size */}
+        <Button
+          aria-controls={menuId}
+          aria-expanded={mobileMenuOpen}
+          aria-haspopup
+          className={styles.toggleButton}
+          fullWidth
+          iconRight={mobileMenuOpen ? <IconAngleUp aria-hidden /> : <IconAngleDown aria-hidden />}
+          onClick={toggleMobileMenu}
+          variant="secondary"
+          type="button"
+        >
+          {toggleButtonLabel}
+        </Button>
+        <ul
+          {...{
+            className: classNames(styles.mainLevelList, mobileMenuOpen && styles.open),
+            'aria-label': toggleButtonLabel,
+            'aria-hidden': isMobile && !mobileMenuOpen,
```

Comment:
This might cause problems for server side rendering. As react requires the initially rendered html is the same as the server side rendered. And because in server we do not know if it's mobile, the html will differ for mobile screens.

Reply:
Good question 👍 
I think it is ok to be the property false as long as it will render the component and won't throw errors. 
I'd guess if we wanted to support fully isMobile property(also in SSR), it should be configurable throughout HDS components.

Reply:
I can test this component in SSR context.

Reply:
Seems to work in hds-next:
<img width="1250" alt="Näyttökuva 2021-8-24 kello 10 40 13" src="https://user-images.githubusercontent.com/1610860/130576963-04122ab2-c8f8-410c-8dc4-024f451c0385.png">